app/routes/bdb/components/StudentContactEditor.tsx

```diff
@@ -0,0 +1,128 @@
+
+export type FormValues = {
+  semesterId: {label: string, value: string};
+  studentContacts: {label: string, value: EntityId}[];
+};
+
+const TypedLegoForm = LegoFinalForm<FormValues>;
+
+const StudentContactEditor = () => {
+  const title = 'Rediger studentkontakter'
+
+  const { companyId } = useParams<{
+    companyId: string;
+  }>();
+
+  const { query, setQueryValue } = useQuery({semesterId: undefined} as {semesterId?: string});
+  const semesterId = query.semesterId;
+
+  const fetching = useAppSelector((state) => state.companies.fetching);
+  const company = useAppSelector((state) =>
+    selectTransformedAdminCompanyById(state, companyId),
+  );
+
+  const semesters = useAppSelector((state) => selectCompanySemesterEntities<CompanySemester>(state));
+  const selectedSemester = semesterId ? semesters[semesterId] : undefined;
+
+  const studentContacts = selectedSemester && company?.studentContacts?.filter(
+    (studentContact) => studentContact.semester.semester == selectedSemester.semester && studentContact.semester.year == selectedSemester.year
+  )
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const onSemesterChange = ({value}: {label: string, value: string}) => {
+    setQueryValue("semesterId")(value);
+  }
+
+  usePreparedEffect(
+    'fetchSemester',
+    () =>
+      companyId && dispatch(fetchSemesters()).then(() => dispatch(fetchAdmin(companyId))),
+    [companyId],
+  );
+
+  if (!company) {
+    return <LoadingPage loading={fetching} />;
+  }
+
+  const onSubmit = async (formContent: FormValues) => {
+    if (!companyId || !semesterId) return;
+
+    const updatedStudentContactsForSemester = formContent.studentContacts.map(studentContact => ({company: Number(companyId), semester: Number(semesterId), user: studentContact.value}));
+
+    const missingStudentContacts = company.studentContacts?.filter(studentContact => studentContact.semester.id != semesterId).map(studentContact => ({company: studentContact.company, user: studentContact.user.id, semester: studentContact.semester.id}));
+
+    const body = {
+      studentContacts: [...updatedStudentContactsForSemester, ...(missingStudentContacts ?? [])],
+      companyId: company.id,
+    };
+
+    await dispatch(editCompany(body));
+    navigate(`/bdb/${companyId}`);
+  };
+
+  const initialValues = (selectedSemester && semesterId) ? {
+      semesterId: {label: semesterToHumanReadable(selectedSemester.semester, selectedSemester.year), value: semesterId},
+      studentContacts: studentContacts?.map(studentContact => ({label: `${studentContact.user.fullName} (${studentContact.user.username})`, value: studentContact.user.id}))
+    } : {}
+
+  return (
+    <Page
+      title={title}
+      back={{ href: `/bdb/${company.id}` }}
+    >
+      <TypedLegoForm
+        onSubmit={onSubmit}
+        initialValues={initialValues}
+      >
+        {({ handleSubmit }) => (
+          <form onSubmit={handleSubmit}>
```

Comment:
```suggestion
          <Form onSubmit={handleSubmit}>
```

https://github.com/webkom/lego-webapp/pull/5258